grails-app/services/de/iteratec/osm/linechart/LineChartTimeSeriesService.groovy

```diff
@@ -0,0 +1,92 @@
+package de.iteratec.osm.linechart
+
+import de.iteratec.osm.csi.Page
+import de.iteratec.osm.measurement.schedule.JobGroup
+import de.iteratec.osm.result.CachedView
+import de.iteratec.osm.result.MeasuredEvent
+import de.iteratec.osm.result.SelectedMeasurand
+import de.iteratec.osm.result.dao.EventResultProjection
+import de.iteratec.osm.result.dao.EventResultQueryBuilder
+import de.iteratec.osm.util.PerformanceLoggingService
+import grails.gorm.transactions.Transactional
+
+import static de.iteratec.osm.util.PerformanceLoggingService.LogLevel.DEBUG
+
+@Transactional
+class LineChartTimeSeriesService {
+
+    PerformanceLoggingService performanceLoggingService
+
+    TimeSeriesChartDTO getTimeSeriesFor(GetLinechartCommand cmd) {
```

Comment:
I would suggest to split this long method into some smaller ones. At least one for getting the result projections and one for building the DTO at the top level. That way it's easier to understand what's happening here.